src/actions/__tests__/importMarkdown.ts

```diff
@@ -0,0 +1,351 @@
+import { HOME_TOKEN } from '../../constants'
+import exportContext from '../../selectors/exportContext'
+import initialState from '../../util/initialState'
+import reducerFlow from '../../util/reducerFlow'
+import importMarkdown, { convertMarkdownToText } from '../importMarkdown'
+
+describe('convertMarkdownToText', () => {
+  it('should import headings', () => {
+    const markdown = `
+# H1
+## H2
+### H3
+#### H4
+##### H5
+###### H6
+`
+
+    expect(convertMarkdownToText(markdown)).toBe(`
+- H1
+  - H2
+    - H3
+      - H4
+        - H5
+          - H6
+`)
+  })
+
+  it('should respect indentation when navigating up and down headings', () => {
+    const markdown = `
+# Heading 1
+a
+## Heading 2
+b
+### Heading 3
+c
+## Heading 2 again
+d
+#### Heading 4
+e
+# Heading 1 again
+f
+### Heading 3 again
+g
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- Heading 1
+  - a
+  - Heading 2
+    - b
+    - Heading 3
+      - c
+  - Heading 2 again
+    - d
+    - Heading 4
+      - e
+- Heading 1 again
+  - f
+  - Heading 3 again
+    - g
+`)
+  })
+
+  it('should handle nested headings', () => {
+    const markdown = `
+- List Item 1
+- ## List Item 2
+- List Item 3
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- List Item 1
+- List Item 2
+- List Item 3
+`)
+  })
+
+  it('should import emphasis', () => {
+    const markdown = `
+*italic*
+_italic_
+**bold**
+__bold__
+***bold italic***
+~~strikethrough~~
+`
+    expect(convertMarkdownToText(markdown)).toBe(`
+- <em>italic</em> <em>italic</em> <strong>bold</strong> <strong>bold</strong> <em><strong>bold italic</strong></em> <del>strikethrough</del>
+`)
+  })
+
+  it('should import lists', () => {
+    const markdown = `
+1. First ordered item
```

Comment:
Could you add the meta attribute `=numbered` to numbered lists? It will be a no-op for now, but at least that information is not lost.

Reply:
We run into the same issue as with tables here, lists are being collapsed and we don't necessarily have an expressive parent that could hold the meta attribute. We can either add `=numbered` to each list item (which would technically allow for a mix of numbered/bulleted thoughts within the same context), or support it via the `=scope` solution we discussed which would probably be the better approach, but unsupported at this point.

Reply:
I see. Maybe we could take a hybrid approach. If the list has other siblings, we separate it out into a `=scope`, otherwise we can apply `=numbered` to the parent.

## Scoped

Markdown:

```
# Heading
a

b

1. c
2. d
3. e
```

Thoughts:

```
- Heading
  - a
  - b
  - ::
    - =scope
    - =numbered
    - c
    - d
    - e
```

## Unscoped

Markdown:

```
# Heading
1. a
2. b
3. c
```

Thoughts:

```
- Heading
  - =scope
  - =numbered
  - a
  - b
  - c
```

The same logic can apply to tables and normal lists as well. What do you think?

Reply:
I believe this should work. I'm seeing two options to implement this, either:
- Check the previous and next blocks for headings and make sure that heading level does not increase.
- Always include `- ::/=scope` while traversing and collapse them afterwards. (This sounds like the cleaner solution)

Reply:
I tend to prefer increasing precision upstream (Option 1) rather than compensating downstream (Option 2). The initial condition is more complex, but it's less spread out. 

Also I'm not sure that detecting a collapsible `::/=scope` is so trivial, especially with nested lists. You have any number of descendants in between. 

One thing to consider is maybe changing `result` to a list of lines that you `.join('\n')` at the end instead of building it as a string. This would make it easier to add/remove `::/scope` in subsequent iterations. Maybe you have a better approach you can think of though.

Reply:
I've implemented collapsing in 0016f3efa906bc800d9cf96ba26803528709f140, running as post-processor. Although this adds some algorithmic complexity, it allows for processing tokens to be as straightforward as possible, allowing for far easier extension with other markdown elements in the future.

Reply:
Yeah, this only confirms what I was saying. You have access to hierarchy information in the first pass that would allow you to more easily detect the beginning and end of ordered lists. By doing it in a second pass, you have to recreate the hierarchy and process a lot of additional output. It would be less complex and more performant to integrate this logic into the first pass.

I suggest creating a separate stack for ordered lists that allows you to insert `::/=scope` as soon as the depth indicates that the ordered list has ended.